File: src/main/java/priv/bajdcc/util/lexer/regex/Repetition.java
package priv.bajdcc.util.lexer.regex;

/**
 * 循环功能
 *
 * @author bajdcc
 */
public class Repetition implements IRegexComponent {
	/**
	 * 循环部件表达式
	 */
	public IRegexComponent component = null;

	/**
	 * 循环次数下限
	 */
	public int iLowerBound = 0;

	/**
	 * 循环次数上限
	 */
	public int iUpperBound = 0;

	public Repetition() {

	}

	public Repetition(IRegexComponent component, int begin, int end) {
		this.component = component;
		iLowerBound = begin;
		iUpperBound = end;
	}

	@Override
	public void visit(IRegexComponentVisitor visitor) {
		visitor.visitBegin(this);
		component.visit(visitor);
		visitor.visitEnd(this);
	}
}

File: src/main/kotlin/com/bajdcc/util/lexer/regex/Repetition.kt
package com.bajdcc.util.lexer.regex

/**
 * 循环功能
 * [component] 循环部件表达式
 * [lowerBound] 循环次数下限
 * [upperBound] 循环次数上限
 * @author bajdcc
 */
data class Repetition (val component: IRegexComponent,
                       var lowerBound: Int,
                       var upperBound: Int) : IRegexComponent {

    override fun visit(visitor: IRegexComponentVisitor) {
        visitor.visitBegin(this)
        component.visit(visitor)
        visitor.visitEnd(this)
    }
}
